### PR TITLE
feat(CB2-6471): suppress cert gen

### PR DESCRIPTION
**Pull request by leifkemp-bjss**
## CB2-6471: Suppress Certificate Generation on Amendments

_Suppresses certification generation on amendments, and adds flag to disable suppression if needed_
[CB2-6471](https://dvsa.atlassian.net/browse/CB2-6471)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Comment by leifkemp-bjss** [comment]
The package lock had to be changed to fix some vulnerabilities, bit of faffing to get it sorted but that's why it's been changed in this PR

**Comment by leifkemp-bjss** [comment]
Scanrepo has also been removed as we don't use it anymore

**Review by kpearmanbjss — DISMISSED** [review]

**Review by gjulien-bjss — APPROVED** [review]

**Review by Ruairidh-BJSS — DISMISSED** [review]

File: src/services/StreamService.ts [hunk 1]
```diff
@@ -28,7 +28,10 @@ class StreamService {
     // Create from a test result with multiple test types, multiple test result with one test type each
     const records: any[] = event.Records.filter((record: DynamoDBRecord) => {
       // Retrieve "INSERT" events
-      return record.eventName === "INSERT" || record.eventName === "MODIFY";
+      return (
+        record.eventName === "INSERT" ||
+        (record.eventName === "MODIFY" && process.env.MODIFY_EVENTS)
```
Comment (aaron-downing) [inline src/services/StreamService.ts:33]: I think for readability it might be better for this env variable to be called `IGNORE_MODIFY_EVENTS`

Comment (leifkemp-bjss) [reply src/services/StreamService.ts]: Done, renamed to PROCESS_MODIFY_EVENTS

File: src/services/StreamService.ts [hunk 2]
```diff
@@ -28,7 +28,10 @@ class StreamService {
     // Create from a test result with multiple test types, multiple test result with one test type each
     const records: any[] = event.Records.filter((record: DynamoDBRecord) => {
       // Retrieve "INSERT" events
-      return record.eventName === "INSERT" || record.eventName === "MODIFY";
+      return (
+        record.eventName === "INSERT" ||
+        (record.eventName === "MODIFY" && process.env.PROCESS_MODIFY_EVENTS)
```
Comment (aaron-downing) [inline src/services/StreamService.ts:33]: Environment variables are explicitly stored as strings and so performing a truthy check on an environment variable is only checking whether the variable is defined. I would suggest creating a function like this and calling it.
```
function processModifyEvents(): boolean {
  if (process.env.PROCESS_MODIFY_EVENTS== 'false') return false;
  if (process.env.PROCESS_MODIFY_EVENTS== 'true') return true;

  throw Error('PROCESS_MODIFY_EVENTS environment variable must be true or false');
}
```

Comment (leifkemp-bjss) [reply src/services/StreamService.ts]: Added

**Review by Ruairidh-BJSS — APPROVED** [review]

**Review by aaron-downing — APPROVED** [review]

**Pull request merged**